src/test/kotlin/ua/marchenko/artauction/artwork/repository/ArtworkRepositoryTest.kt

```diff
@@ -0,0 +1,142 @@
+        // GIVEN
+        val savedArtwork = artworkRepository.save(MongoArtwork.random(id = null))
+
+        // WHEN
+        val result = artworkRepository.findById(savedArtwork.id.toString())
+
+        // THEN
+        assertEquals(savedArtwork.id, result?.id)
+    }
+
+    @Test
+    fun `should return null when there is no artwork with this id`() {
+        // WHEN
+        val result = artworkRepository.findById(getRandomObjectId().toString())
+
+        // THEN
+        assertNull(result, "Found artwork must be null")
+    }
+
+    @Test
+    fun `should return artwork with artist when artwork with this id exists`() {
+        // GIVEN
+        val savedArtist = userRepository.save(MongoUser.random(id = null))
+        val savedArtwork = artworkRepository.save(MongoArtwork.random(id = null, artistId = savedArtist.id.toString()))
+
+        // WHEN
+        val result = artworkRepository.findFullById(savedArtwork.id.toString())
+
+        // THEN
+        assertEquals(savedArtist.name, result?.artist?.name)
+    }
+
+    @Test
+    fun `should return null when artist of artwork doesnt exist in db`() {
+        // GIVEN
+        val savedArtwork = artworkRepository.save(MongoArtwork.random(id = null))
+
+        // WHEN
+        val result = artworkRepository.findFullById(savedArtwork.id.toString())
+
+        // THEN
+        assertNull(result, "Found artwork must be null")
+    }
+
+    @Test
+    fun `should return all artworks with artist when they are exists`() {
+        // GIVEN
+        val savedArtist = userRepository.save(MongoUser.random(id = null))
+        val artworks = listOf(
+            MongoArtwork.random(artistId = savedArtist.id.toString()),
+            MongoArtwork.random(artistId = savedArtist.id.toString())
+        )
+        artworks.forEach { artwork -> artworkRepository.save(artwork) }
+
+        // WHEN
+        val result = artworkRepository.findFullAll()
+
+        // THEN
+        assertTrue(result.size >= artworks.size, "Size of list must be at least ${result.size}")
+        artworks.forEach { artwork ->
+            assertTrue(
+                result.any { it.title == artwork.title && it.artist?.name == savedArtist.name },
+                "Artwork with title ${artwork.title} and artist ${savedArtist.name} must be found"
+            )
+        }
+    }
+
+    @Test
+    fun `should return all artworks when they are exists`() {
+        // GIVEN
+        val artworks = listOf(MongoArtwork.random(id = null), MongoArtwork.random(id = null))
+        artworks.forEach { artwork -> artworkRepository.save(artwork) }
+
+        // WHEN
+        val result = artworkRepository.findAll()
+
+        // THEN
+        assertTrue(result.size >= artworks.size, "Size of list must be at least ${result.size}")
+        artworks.forEach { artwork ->
+            assertTrue(result.any { it.title == artwork.title }, "Artwork with title ${artwork.title} must be found")
+        }
+    }
```

Comment:
When this test can possibly fail?
Seems like you testing save here

Reply:
if saved artwork would not be found in returned list